### ISSUE TITLE
Increase the usage of compound assignment operators

### ISSUE THREAD

**Issue by elfring** [FPGAwars/icestudio#558]
:eyes: Some source code analysis tools can help to find opportunities for improving software components.
:thought_balloon: I propose to [increase the usage of compound operators](https://262.ecma-international.org/12.0/#sec-assignment-operators "Description for assignment operators") accordingly.

Would you like to integrate anything from a transformation result which can be generated by a command like the following?
(:point_right: Please check also for questionable change suggestions because of an evolving search pattern.)
```
lokal$ perl -p -i.orig -0777 -e 's#\b(?<target>\S+)\s*=\s*\k<target>[ \t]*(?<operator>[+\-/%^]|\*\*?|&&?|\|\|?|\?\?|<<|>>>?)#$+{target} $+{operator}=#gm' $(find ~/Projekte/icestudio/lokal -name '*.js')
```

:crystal_ball: How will development interests evolve further also according to update candidates in 40 lines of this software?

**Comment by elfring** [comment]
>… - your goal is not very clear, …

:thought_balloon: Would you dare to perform a source code search and corresponding replacements in another development repository with the help of the shown Perl command example?

**Comment by cavearr** [comment]
Hi ! i'm closing this issue because is not an issue as is.

**Comment by elfring** [comment]
:crystal_ball: Would you ever become interested in the presented source code transformation possibilities?

**Comment by cavearr** [comment]
Yes of course, any optimization or improvement is welcome. Right now I have been refactoring and improving the code for several months. I will soon deploy a new version that includes improvements like the ones you propose. In any case, if you have ideas of specific things in which you can demonstrate that in specific parts of the code the readability or performance is improved (with data and measurements that justify it) we will always be open to you making PRs.
